lib/midcom/helper/datamanager2/widget/tinymce.php

```diff
@@ -286,49 +287,10 @@ private function _get_imagetools_configuration()
         return <<<EOT
 imagetools_toolbar: "editimage imageoptions",
 imagetools_cors_hosts: ['{$hostname}'],
-setup: function(editor) 
-{
-    editor.on('click', function(e) 
-    {
-        var node = tinymce.activeEditor.selection.getNode();
-        if(node.hasAttribute("src"))
-        {
-            original = node.src.split("/").pop();
-        }
-    });
-},
-images_upload_handler: function (blobInfo, success, failure)
-{
-    var xhr, formData;
-    xhr = new XMLHttpRequest();
-    xhr.withCredentials = true;
-    xhr.open('POST', '{$url}');
-
-    xhr.onload = function() 
-    {
-        var json;
-        if(xhr.status != 200) 
-        {
-            failure('HTTP Error: ' + xhr.status);
-            return;
-        }
-
-        json = JSON.parse(xhr.responseText);
-            
-        if(!json || typeof json.location != 'string') 
-        {
-            failure('Invalid JSON: ' + xhr.responseText);
-            return;
-        }
-               
-        success(json.location);
-    };
-             
-    var name = original.split(".").shift() + "." + blobInfo.filename().split(".").pop();
-    formData = new FormData();
-    formData.append('file', blobInfo.blob(), name);
-    xhr.send(formData);
-},
+url: '{$url}',
+original: "",
+setup: imagetools_functions.setup,
+images_upload_handler: imagetools_functions.images_upload_handler,
```

Comment:
I think it would be cleaner & safer to do something like 

```javascript
images_upload_handler: imagetools_functions.images_upload_handler('{$url}')
```

and then in `imagetools.js`:


```javascript
images_upload_handler: function(url)
{
    return function (blobInfo, success, failure) {
        //...
    }
}
```